src/Umbraco.Web/Editors/EntityController.cs

```diff
@@ -235,13 +245,61 @@ private string ParseXPathQuery(string query, int id)
                 },
                 publishedContentExists: i => Umbraco.TypedContent(i) != null);
         }
-        
+
+        #region GetById
+
+        /// <summary>
+        /// Gets an entity by it's id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
         public EntityBasic GetById(int id, UmbracoEntityTypes type)
         {
             return GetResultForId(id, type);
         }
 
-        public IEnumerable<EntityBasic> GetByIds([FromUri]int[] ids, UmbracoEntityTypes type)
+        /// <summary>
+        /// Gets an entity by it's key
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        public EntityBasic GetById(Guid id, UmbracoEntityTypes type)
+        {
+            return GetResultForKey(id, type);
+        }
+
+        /// <summary>
+        /// Gets an entity by it's UDI
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        public EntityBasic GetById(Udi id, UmbracoEntityTypes type)
+        {
+            var guidUdi = id as GuidUdi;
+            if (guidUdi != null)
+            {
+                return GetResultForKey(guidUdi.Guid, type);
+            }
+            throw new HttpResponseException(HttpStatusCode.NotFound);

```

Comment:
Not IMO - returning more than you need from an endpoint can be a security risk, give as little away as possible, only what is required. In this case, nothing is found, this endpoint doesn't support that type.